src/Microsoft.ML.Tokenizers/Model/EnglishRoberta.cs

```diff
@@ -538,26 +531,19 @@ private static int GetByteToUnicode(out IReadOnlyDictionary<char, char> byteToUn
         }
 
         /// <summary>
-        /// Encode a token into BPE-ed Ids. E.g., "playing" into ["play", "ing"].
+        /// Encode a token into BPE-ed sub-tokens. E.g., "playing" into ["play", "ing"].
         /// </summary>
-        /// <param name="token">The token to encode.</param>
-        /// <param name="ids">The list of Ids to encode the token into.</param>
-        /// <returns>The number of encoded ids.</returns>
-        private int EncodeToIds(Span<char> token, IList<int>? ids)
+        private List<Token> EncodeToTokens(Span<char> token, Span<int> indexMapping)
         {
             if (token.Length == 0)
             {
-                return 0;
+                return Bpe.EmptyTokensList;
             }
 
             if (token.Length == 1)
             {
-                if (ids is not null)
-                {
-                    ids.Add(_vocab[_charToString[token[0]]]);
-                }
-
-                return 1;
+                string tokenValue = _charToString[token[0]];
```

Comment:
token[0] is guaranteed to be < _charToString.Length?

Reply:
Yes it s guaranteed. This method `EncodeToTokens` is called from Encode method which filter the token's characters before using it by doing  `if (_byteToUnicode.TryGetValue(text[i], out var value))`. `_byteToUnicode` and `_charToString` contain the same set of keys.